docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
+
+The add income/expense command has 2 compulsory arguments `/t` and `/a` and 1 optional argument `/r`.
+
+Example:
+```
+add income /a 100 /t salary /r 30
+```
+Below are the steps that shows the implementation of add income/expense.
+#### Step 1
+An instantiated AddCashflowCommand class gets the instance of CashflowList.
+
+This allows the AddCashflowCommand instance to access the methods of CashflowList.
+#### Step 2
+The AddCashflowCommand instance then calls addIncome() or addExpense(), depending on what `category` is initialised as.
+
+addIncome() or addExpense() instantiates an Income or Expense object respectively.
+
+Example:
+```
+switch (category) {
+        case INCOME:
+            cashflowList.addIncome(amount, incomeType, recur);
+            break;
+        case EXPENSE:
+            cashflowList.addExpense(amount, expenseType, recur);
+            break;
+        default:
+            ui.showMessage("Unidentified entry.");
+            break;
+        }
+```
+#### Step 3
+The instantiated income/expense then updates the overall balance through addIncomeValue() or addExpenseValue().
+
+The income/expense object is also added to the list in Cashflowlist which contains all incomes/expenses.
+#### Step 4
+The added income/expense is then displayed to the user through the Ui.
+
+#### Diagrams
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/CashflowClassDiagram.png)
+
+Given below is the sequence diagram showing the add income/expense mechanism:
+![](images/AddCashflowSequence.png)
+
+
+### Budget Feature
+
+This feature has 5 functions, `set`, `update`, `delete`, `reset`, and `view`.
+
+![](images/Budget.png)
+
+The BudgetCommand will execute the appropriate command and print through `Budget.java` and prints any message to the user through `Ui.java`.
+
+#### Set and update budget:
+
+Example:
+```
+budget set /b 500
+budget update /b 1000
+```
+The '/b' is followed by the budget amount. The first line will set the budget by calling `setBudget(500)` method in `Budget.java`.
+The second line updates the budget by adding or subtracting the difference between the initial and updated amount to the 
+initial and current budget. This is done through `updateBudget(500)` method in `Budget.java`. Both functions can be seen 
+in the diagram above
+
+#### Delete budget:
+
+![](images/deleteBudget.png)
+
+The budget will be deleted by setting the initial and current budget to 0 through the `deleteBudget()` method in `Budget.java`.
+
+Example: `budget delete`
+
+#### Reset budget:
+
+![](images/resetBudget.png)
+
+The budget will be reset by resetting the current budget to the initial budget through the `resetBudget()` method in `Budget.java`.
+
+Example: `budget reset`
+
+#### View budget:
+
+![](images/viewBudget.png)
+
+The current budget will be shown to the user through the `Ui`.
+
+Example: `budget view`
+
 
 ## Product scope
 ### Target user profile
```

Comment:
May want to add more User Stories to accurately reflect your actual feature implementation in each version.